src/internal/packager/images/pull.go

```diff
@@ -486,8 +486,11 @@ func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Im
 	}
 
 	spinner.Updatef("Preparing image %s", src)
-	imageCachePath := filepath.Join(config.GetAbsCachePath(), layout.ImagesDir)
-	img = cache.Image(img, cache.NewFilesystemCache(imageCachePath))
+
+	if !strings.HasSuffix(src, ".sig") && !strings.HasSuffix(src, ".att") {
```

Comment:
At this point you have access to the `img` object and it's `MediaType()` which you should be able to check to see if this is actually an image manifest or not (instead of fuzzily checking the "extension")